blog-posts/common-approaches-to-handling-subscriptions/common-approaches-to-handling-subscriptions.md

```diff
@@ -0,0 +1,178 @@
+
+Consider the following code:
+
+```javascript
+import { interval } from 'rxjs';
+import { take } from 'rxjs/operators';
+
+const intervalCount = interval(1000);
+const takeObservable = intervalCount.pipe(take(2));
+takeObservable.subscribe(x => console.log(x));
+```
+
+When you run this, you should see the following:
+
+```bash
+0
+1
+```
+
+Now what if you changed that same code to the following:
+
+```javascript
+import { interval } from 'rxjs';
+import { take } from 'rxjs/operators';
+
+const intervalCount = interval(1000);
+const takeObservable = intervalCount.pipe(take(10));
+takeObservable.subscribe(x => console.log(x));
+```
+
+When you run this, you should see the same as before but the count goes from 0 to 9.
+
+So what's happening? The take operator just controls the execution flow so that the number you pass in determines how many times it emits a value before completing. You don't have to worry about a memory leak here because the completion formally stops the execution flow here.
+
+In addition to the `take` operator there are multiple other examples of ways to do this behavior.
+
+Some include the following:
+
+- [takeWhile](https://rxjs.dev/api/operators/takeWhile)
+- [takeUntil](https://rxjs.dev/api/operators/takeUntil)
+- [takeLast](https://rxjs.dev/api/operators/takeLast)
+- [first](https://rxjs.dev/api/operators/first)
+
+The important thing about this behavior is just that you're letting RxJS handle the stream for you. This lets you write clean code that is easily maintainable.
+
+## Combining Subscriptions
+
+Another common pattern that you run across is when you multiple observables, and want to manage their subscriptions together.
+
+Consider the following code:
+
+```javascript
+import { Subscription, of } from 'rxjs';
+
+// create a subscription object
+const subs = new Subscription();
+
+// create observables
+const value$ = of(1, 2, 3, 4);
+const anotherValue$ = of(true);
+
+// subscribe to observables and add to subscription
+subs.add(value$.subscribe(x => console.log(x)));
+subs.add(anotherValue$.subscribe(x => console.log(x)));
+
+// calling subs.unsubscribe() will unsubscribe from all sub
+subs.unsubscribe();
+```
+
+> special thanks to my friend [Tim Deschryver](https://twitter.com/tim_deschryver) for this specific example
+
+In this example you see that we define an instance of a [Subscription](https://rxjs.dev/api/index/class/Subscription) that we add two observables to. The `Subscription` class enables you to wrap your subscriptions in one resource. When you're ready to dispose of your application, you can just call a singular `unsubscribe` and execution in all of the observables wrapped will be properly stopped.
+
+This pattern is particularly useful when you have multiple observables in a component that you want to manage together. It makes the implementation cleaner and easier to maintain.
+
+## Using Frameworks and the Angular async pipe
```

Comment:
I personally think it would be better to have this in a seperate blog article because it introduces concepts for angular and NgRx which might be confusing for readers. But we can discuss the seperatly via twitter, it's propably easier

Reply:
Agree, taking this section out 👍